johnsonandjohnson/Bodiless-JS
# feat(richtext): Refactor the RichText API

## Changes
* Makes the RichText Component take a design prop following the DesignAPI
* Change internal to have use components and not richTextItems (removed the RichTextItems)
* Create functionality in RichText that makes it so that we have a default for component with default configuration
* Update documantation 

## Test Instructions


* In general regression testing of the richtext editor needs to be done. (nothing in simple and basic should have changed)
* The technical review should play around with using the following function in the test site in the design that is pass to the richtext editor
** asBlock (should expect that inline items can be set inside of it)
** withButton (should show a button that toggles the formating)
** withShortcutKey (should allow the ctrl-KEY should toggle the format)
** other style related HOC should style appropriately
** startWith (should change the underline component)
There are examples to review of all of this in the full-featured editor

## Related Issues

Closes #10 

BREAKING CHANGE: this changes the API for the richtext Editor
